sdk/java-sdk/src/main/java/kalix/javasdk/action/ActionProvider.java

```diff
@@ -40,7 +38,7 @@
 
   Descriptors.FileDescriptor[] additionalDescriptors();
 
-  default Map<Class<?>, Serializer> additionalSerializers() {
-    return Collections.emptyMap();
+  default Optional<MessageCodec> alternativeCodec() {
+    return Optional.empty();
```

Comment:
We will use `AnySupport` when this is empty. Otherwise, we use the alternative codec defined by the provider.